cpp/ql/test/library-tests/dataflow/external-models/validatemodels.expected

```diff
@@ -1,8 +1,29 @@
+| Dubious member name "operator +=" in summary model. |
+| Dubious member name "operator BSTR" in summary model. |
+| Dubious member name "operator LPCSTR" in summary model. |
+| Dubious member name "operator LPSAFEARRAY" in summary model. |
+| Dubious member name "operator LPSTR" in summary model. |
+| Dubious member name "operator LPWSTR" in summary model. |
+| Dubious member name "operator PCXSTR" in summary model. |
+| Dubious member name "operator&" in summary model. |
 | Dubious member name "operator*" in summary model. |
+| Dubious member name "operator+=" in summary model. |
```

Comment:
How come we have an `operator +=` and an `operator+=`?

I feel that we should consider either cleaning up these names, or loosening the validator?  I'm not sure if there are any real concerned about ambiguity caused by strange characters here.

Reply:
The validation should definitely be loosened. It's a validation shared between all languages, and because of operator overloading and varies derived types (such as `*` int `int*` and `&` in `int&`) C++ simply has more complex type and function names.

Good thing about `operator+=` vs. `operator +=`. One of those is a bug as the correct one is whatever is used by `Function.getName` and I can't actually remember which one is the correct one off the top of my head. But I'll check this and push a fix to the wrong ones.

I think we should delay loosening the validation since I don't want to add more changes than what is necessary here.

Reply:
Testcase for this added (or rather: testcase _fixed_) in 3abb9049bb1fb9d3aa8031b42c4621ad0de7224e, and the incorrect model is fixed in c3086d4ecda3b5c8ba986fb89c9c7ad408daad3b. Sadly, there `MISSING` result can't easily be fixed right now as it likely requires some extractor changes.